### DISCUSSION TITLE
Is there a way to integrate libVLC with Tauri?

### DISCUSSION THREAD

**Discussion by hxhxhx88** [tauri-apps/tauri#7895] [Q&A]
I am using Tauri to build a cross-platform desktop application. Most parts of my application are written in React and are working happily with Tauri. However, the video player part causes some trouble.

The HTML `video` tag can play only a limited set of video formats and codecs, and I would like to use [libVLC](https://www.videolan.org/vlc/libvlc.html) to overcome such limit. Furthermore, I would like to render the video only in a subarea of the window.

For now I am trying something like:
```rust
use vlc::{Instance, Media, MediaPlayer};
fn main() {
    tauri::Builder::default()
        .on_page_load(|window, _| {
            let instance = Instance::new().unwrap();
            let md = Media::new_path(&instance, "/my/video.mp4").unwrap();
            let mdp = MediaPlayer::new(&instance).unwrap();
            mdp.set_media(&md);

            // set the window handle to the vlc media player for rendering the video
            let handle = window.ns_window().unwrap();
            mdp.set_nsobject(handle);

            // does not work.
            mdp.play().unwrap();
        })
        .invoke_handler(tauri::generate_handler![greet])
        .run(tauri::generate_context!())
        .expect("error while running tauri application");
}
```

However, the above code does not work with the log:
```
cache_read stream error: cannot pre fill buffer
mjpeg demux error: cannot peek
```

The video playing part does work if running directly within `main()`:
```rust
use vlc::{Instance, Media, MediaPlayer};
fn main() {
    let instance = Instance::new().unwrap();
    let md = Media::new_path(&instance, "/my/video.mp4").unwrap();
    let mdp = MediaPlayer::new(&instance).unwrap();
    mdp.set_media(&md);

    // does work although you can only hear the sound
    mdp.play().unwrap();
}
```

Besides, I failed to find a way to create a subview within the window. Calling `window.ns_window()` presumably will render the video into the whole window.

My code follows this [example of video playing in Rust with libVLC](https://github.com/fltk-rs/demos/blob/master/libvlc/src/main.rs).

Therefore, my question is:
- is it possible to somehow place a subview in a window, and get the handle to that subview?
- how to fix `mdp.play()` not working within Tauri issue?

Thanks!

**Comment by Ar-mane** [comment]
i reaally need this
